官方脚手架demo npm安装方式无法运行

<!-- 请不要删除自动生成的 Issue 标签 -->
<!-- 请不要删除自动生成的 Issue 标签 -->
### 相关平台
微信小程序


**小程序基础库: 3.5.5**
**使用框架: Vue 3**

### 复现步骤
1.安装taro4.0.6
2.新建Vue3   JavaScript   Npm   Vite   Github   vue3-NutUI
3.npm install安装依赖 依赖冲突报错

### 期望结果
npm和yarn都能正常运行demo

### 实际结果
yarn可以正常装包运行demo，npm装包依赖冲突无法运行

### 环境信息
```
Taro CLI 4.0.6 environment info:
    System:
      OS: macOS 15.0
      Shell: 3.2.57 - /bin/sh
    Binaries:
      Node: 22.9.0 - /usr/local/bin/node
      Yarn: 1.22.22 - /usr/local/bin/yarn
      npm: 10.9.0 - /usr/local/bin/npm
    npmPackages:
      @tarojs/cli: 4.0.6 => 4.0.6 
      @tarojs/components: 4.0.6 => 4.0.6 
      @tarojs/helper: 4.0.6 => 4.0.6 
      @tarojs/plugin-framework-vue3: 4.0.6 => 4.0.6 
      @tarojs/plugin-html: 4.0.6 => 4.0.6 
      @tarojs/plugin-platform-alipay: 4.0.6 => 4.0.6 
      @tarojs/plugin-platform-h5: 4.0.6 => 4.0.6 
      @tarojs/plugin-platform-jd: 4.0.6 => 4.0.6 
      @tarojs/plugin-platform-qq: 4.0.6 => 4.0.6 
      @tarojs/plugin-platform-swan: 4.0.6 => 4.0.6 
      @tarojs/plugin-platform-tt: 4.0.6 => 4.0.6 
      @tarojs/plugin-platform-weapp: 4.0.6 => 4.0.6 
      @tarojs/runtime: 4.0.6 => 4.0.6 
      @tarojs/shared: 4.0.6 => 4.0.6 
      @tarojs/taro: 4.0.6 => 4.0.6 
      @tarojs/vite-runner: 4.0.6 => 4.0.6 
      babel-preset-taro: 4.0.6 => 4.0.6 
      eslint-config-taro: 4.0.6 => 4.0.6 
```

<!-- generated by taro-issues. 请勿修改或删除此行注释 --><!--labels=T-weapp,V-3,F-vue3-->

通过taro init demo初始化项目，选择如下：
￼
<img width="485" alt="Pasted Graphic" src="https://github.com/user-attachments/assets/ad755a0d-3c55-4174-8a2b-5e97f45ad51f">

报错如下，@tarojs/plugin-framework-vue3需要 @vitejs/plugin-vue 的5.x版本。
``` bash
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @tarojs/plugin-framework-vue3@4.0.6
npm error Found: @vitejs/plugin-vue@4.6.2
npm error node_modules/@vitejs/plugin-vue
npm error   dev @vitejs/plugin-vue@"^4.0.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peerOptional @vitejs/plugin-vue@"^5" from @tarojs/plugin-framework-vue3@4.0.6
npm error node_modules/@tarojs/plugin-framework-vue3
npm error   @tarojs/plugin-framework-vue3@"4.0.6" from the root project
npm error
npm error Conflicting peer dependency: @vitejs/plugin-vue@5.1.4
npm error node_modules/@vitejs/plugin-vue
npm error   peerOptional @vitejs/plugin-vue@"^5" from @tarojs/plugin-framework-vue3@4.0.6
npm error   node_modules/@tarojs/plugin-framework-vue3
npm error     @tarojs/plugin-framework-vue3@"4.0.6" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

尝试升级 @vitejs/plugin-vue 5.x, 报错vite版本与 @vitejs/plugin-vue 5.x冲突。
``` bash
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: demo@1.0.0
npm error Found: vite@4.5.5
npm error node_modules/vite
npm error   dev vite@"^4.2.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peer vite@"^5.0.0" from @vitejs/plugin-vue@5.1.4
npm error node_modules/@vitejs/plugin-vue
npm error   dev @vitejs/plugin-vue@"^5" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

找到了以前的相同问题，issue #16303，3个月前的bug了。

--- a/packages/eslint-config-taro/package.json
+++ b/packages/eslint-config-taro/package.json
@@ -35,7 +35,7 @@
     "eslint": "^8",
     "eslint-plugin-react": "^7.33.2",
     "eslint-plugin-react-hooks": "^4.4.0",
-    "eslint-plugin-vue": "^9.17.0"
+    "eslint-plugin-vue": "^8.0.0"
   },
   "peerDependenciesMeta": {
     "eslint-plugin-react": {

--- a/packages/taro-framework-vue3/package.json
+++ b/packages/taro-framework-vue3/package.json
@@ -41,7 +41,7 @@
     "@tarojs/taro": "workspace:*",
     "vue-loader": "^17.0.0",
     "@vue/compiler-core": "^3.4.25",
-    "vite": "^4.2.0",
+    "vite": "^5.2.0",
     "webpack": "5.91.0",
     "vue": "^3.4.25"
   },
@@ -53,7 +53,7 @@
     "@vitejs/plugin-vue": "^5",
     "@vitejs/plugin-vue-jsx": "^3",
     "vue-loader": "^17.0.0",
-    "vite": "^4",
+    "vite": "^5",
     "vue": "^3",
     "webpack": "^5"
   },

--- a/packages/taro-framework-vue3/package.json
+++ b/packages/taro-framework-vue3/package.json
@@ -41,7 +41,7 @@
     "@tarojs/taro": "workspace:*",
     "vue-loader": "^17.0.0",
     "@vue/compiler-core": "^3.4.25",
-    "vite": "^5.2.0",
+    "vite": "^4.2.0",
     "webpack": "5.91.0",
     "vue": "^3.4.25"
   },
@@ -50,10 +50,10 @@
     "@tarojs/runtime": "workspace:*",
     "@tarojs/runner-utils": "workspace:*",
     "@tarojs/shared": "workspace:*",
-    "@vitejs/plugin-vue": "^5",
+    "@vitejs/plugin-vue": "^4",
     "@vitejs/plugin-vue-jsx": "^3",
     "vue-loader": "^17.0.0",
-    "vite": "^5",
+    "vite": "^4",
     "vue": "^3",
     "webpack": "^5"
   },

